File: jdroid-java-googleplay-publisher/src/main/java/com/jdroid/android/googleplay/publisher/TrackType.java
package com.jdroid.android.googleplay.publisher;

public enum TrackType {
	
	INTERNAL("internal"),
	ALPHA("alpha"),
	BETA("beta"),
	PRODUCTION("production");
	
	private String key;
	
	TrackType(String key) {
		this.key = key;
	}
	
	public String getKey() {
		return key;
	}
	
	public static TrackType findByKey(String key, TrackType defaulTrackType) {
		for (TrackType each : TrackType.values()) {
			if (each.getKey().equals(key)) {
				return each;
			}
		}
		return defaulTrackType;
	}

	public static TrackType findByKey(String key) {
		return findByKey(key, null);
	}
}

File: google-play-api-kotlin/src/main/java/com/dipien/google/play/api/TrackType.kt
package com.dipien.google.play.api

enum class TrackType(val key: String) {

    INTERNAL("internal"),
    ALPHA("alpha"),
    BETA("beta"),
    PRODUCTION("production");

    companion object {
        fun findByKey(key: String?, defaulTrackType: TrackType? = null): TrackType? {
            for (each in values()) {
                if (each.key == key) {
                    return each
                }
            }
            return defaulTrackType
        }
    }
}
